wallet_ui/components/panels/CreateCanister.tsx

```diff
@@ -88,43 +87,42 @@ export function CreateCanisterDialog(props: {
   }
   function increaseInput() {
     setCount(count + 1);
-    setMoreControllers(prev => [...prev, ""]);
+    setControllers(prev => [...prev, ""]);
   }
 
   function handleInputChange(ind: number, ev: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
-    const newInput = moreControllers.map((controller : string, i : number) => {
+    const newInput = controllersState.map((controller : string, i : number) => {
       if(ind === i) {
         return ev.target.value;
       }
       return controller;
     })
-    setMoreControllers(newInput);
+
+    setControllers(newInput);
     try {
       Principal.fromText(ev.target.value);
-      setError2(false);
+      setError(p => {
+        let newErr = p;
+        p[ind] = false;
+        return newErr;
+      });
     } catch {
-      setError2(true);
+      setError(p => {
+        let newErr = p;
+        p[ind] = true;
+        return newErr;
+      });
     }
   }
 
   function deleteInput(ind : number) {
-    const result = moreControllers.filter((con : string, i: number) => {
+    const result = controllersState.filter((con : string, i: number) => {
```

Comment:
To remove a single entry from an array, the canonical method is to use splice. Filter isn't wrong per se, but here is another example

```suggestion
    const newControllers = [...controllers];
    newControllers.splice(ind, 1);
    setControllers(newControllers);
```